### DISCUSSION TITLE
Is it possible to have multiple completions for one command?

### DISCUSSION THREAD

**Discussion by wsw70** [fish-shell/fish-shell#10837] [Q&A]
Imagine I have a command of the type

    mycommand parameter1 parameter2

I would like to enable completion for `parameter1` and once it is chosen, completion for `parameter2`. In other words, the sequence could be 

```
$ mycommand <TAB> → this lists possible choices for parameter1, among them "start"
$ mycommand start <TAB> → this lists possible choices for parameter 2, generated with the predicate "start" (in other words, parameter2 = f(parameter1)
```

Is this possible?

**Comment by zanchey** [comment]
Yes, this is the subcommand pattern. The `wg` completions are a fairly simple implementation if you want to start with that.

**Reply by wsw70** [reply]
> this is the subcommand pattern

Ah, sorry, I forgot to mention that I specifically do not want to use the subcommand approach. I started with that but it complexifies my code a lot - I now use two positional parameters.

**Reply by zanchey** [reply]
It's not clear to me how that differs, but have a look at how `__fish_use_subcommand` or `__fish_seen_subcommand_from` work and you should be able to translate it into your approach.
